### PR TITLE
unified transition resource

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/2023

Requires #581 (for nothing special, has been forked from it because it had more components migrated)

These changes are motivated by the issues we are finding with dependent resources in operatorkit https://github.com/giantswarm/operatorkit/issues/139 having a single resource for both the legacy and the new cloudformation code solves the dependency problems, we can decide when to create or delete each of the components.

I've tested locally and works fine for both 0.1.0 and 0.2.0, let me know WDYT. The downside is that the single resource is more complex, I think we can iterate to make things more manageable. The old cloudformationv{1,2} are removed to prevent test failures when the template would evolve. I haven't put in place all the existing unit tests for resource-related methods, if we agree in this approach I'll move them here.

### CODE REVIEW

File: service/resource/legacyv2/resource.go [hunk 1]
```diff
@@ -207,12 +215,31 @@ func (s *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desire
 
 	s.logger.Log("info", fmt.Sprintf("cluster '%s' processed", keyv2.ClusterID(cluster)))
 
-	return &framework.Patch{}, nil
+	patch := framework.NewPatch()
+
+	// cloudformation logic
```
Comment (rossf7) [inline service/resource/legacyv2/resource.go]: I think more comments here would be good. To mention that the resources are combined during the transition due to the deletion order problem and that they will be separated later.

File: service/resource/legacyv2/resource.go [hunk 2]
```diff
@@ -180,22 +190,20 @@ type Resource struct {
...
 
-type Event struct {
-	Type   string
-	Object *v1alpha1.AWSConfig
-}
-
 // NewUpdatePatch is called upon observed custom object change. It creates the
 // AWS resources for the cluster.
 func (s *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
+	s.logger.Log("info", "in NewUpdatePatch")
```
Comment (rossf7) [inline service/resource/legacyv2/resource.go]: Are these logs for debugging or will they be kept?

File: service/resource/s3objectv2/desired.go [hunk 3]
```diff
@@ -27,6 +27,10 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	}
 
 	kmsArn, err := r.awsService.GetKeyArn(clusterID)
+	if IsKeyNotFound(err) {
+		// we can get here during deletion, if he key is already deleted we can safely exit
```
Comment (rossf7) [inline service/resource/s3objectv2/desired.go]: Good this is handled and commented. Just a typo on "the key" and can you make it a sentence.

File: service/resource/legacyv2/resource.go (hunk 2 above)
Comment (fgimenez) [reply service/resource/legacyv2/resource.go]: Only for making clear the code flow during debug, removed.

File: service/resource/legacyv2/resource.go (hunk 1 above)
Comment (fgimenez) [reply service/resource/legacyv2/resource.go]: Indeed, I've extended the existing ones, please take a look.

File: service/resource/s3objectv2/desired.go (hunk 3 above)
Comment (fgimenez) [reply service/resource/s3objectv2/desired.go]: Sure, done

**Pull request merged**